src/PrestaShopBundle/Service/Database/DoctrineRuntimeEntityMapper.php

```diff
@@ -0,0 +1,141 @@
+ * during runtime. It automatically scans the entity folder (annotation only for
+ * now) and adds them to the doctrine mapping.
+ */
+class DoctrineRuntimeEntityMapper
+{
+    /**
+     * @var Reader
+     */
+    private $reader;
+
+    /**
+     * @var MappingDriverChain
+     */
+    private $mappingDriverChain;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @param Reader $reader
+     * @param MappingDriverChain $mappingDriverChain
+     * @param Configuration $configuration
+     */
+    public function __construct(
+        Reader $reader,
+        MappingDriverChain $mappingDriverChain,
+        Configuration $configuration
+    ) {
+        $this->reader = $reader;
+        $this->mappingDriverChain = $mappingDriverChain;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * @param string $entityFolder
+     * @param string $entityAlias
+     *
+     * @throws EntityFolderNotFoundException
+     * @throws EntityNamespaceNotFoundException
+     */
+    public function addDoctrineMapping(string $entityFolder, $entityAlias = '')
+    {
+        if (!is_dir($entityFolder)) {
+            throw new EntityFolderNotFoundException(sprintf(
+                'Cannot find entity folder %s',
+                $entityFolder
+            ));
+        }
+
+        $entityNamespace = $this->getEntityNamespace($entityFolder);
+        $driver = $this->createAnnotationDriver($entityFolder);
+        $this->mappingDriverChain->addDriver($driver, $entityNamespace);
+        if (!empty($entityAlias)) {
+            $this->configuration->addEntityNamespace($entityAlias, $entityFolder);
+        }
+    }
+
+    /**
+     * This method is derived from DoctrineOrmMappingsPass::createAnnotationMappingDriver, except we only focus on
+     * creating the driver here. And (as in ModulesDoctrineCompilerPass) we make sure that the driver will ignore
+     * the index.php file that may be present in the entity folder.
+     *
+     * @param $entityFolder
+     *
+     * @return AnnotationDriver
+     */
+    private function createAnnotationDriver($entityFolder)
+    {
+        $driver = new AnnotationDriver($this->reader, [$entityFolder]);
+        $indexFile = $entityFolder . '/index.php';
+        if (file_exists($indexFile)) {
+            $driver->addExcludePaths([$indexFile]);
+        }
+
+        return $driver;
+    }
+
+    /**
+     * @param string $entityFolder
+     *
+     * @return string
+     *
+     * @throws EntityNamespaceNotFoundException
+     */
+    private function getEntityNamespace(string $entityFolder)
+    {
+        $finder = new Finder();
+        $finder->files()->in($entityFolder)->name('*.php');
+        foreach ($finder as $phpFile) {
+            $phpContent = file_get_contents($phpFile->getRealPath());
+            if (preg_match('~namespace[ \t]+(.+)[ \t]*;~Um', $phpContent, $matches)) {
```

Comment:
idem for this regex (use constant or share it if it is shareable)

Reply:
Yes, actually I think we shoud have a class/service dedicated to this, since @sowbiba also needed this in one of his PR I thought about adding it in another PR but I can add it now